### PR TITLE
Add a note about flushing D-Cache on RT10xx

### PR DESCRIPTION
**Pull request by noahp**

### CODE REVIEW

File: docs/mcu/arm-mcuxpresso-guide.mdx [hunk 1]
```diff
@@ -357,3 +357,30 @@ implementation.
...
+non-initialized (`.noinit`) RAM when the system encounters a crash. If the
+RAM-backed coredump storage Memfault component is used, that data also should be
+located in a non-initialized section of RAM.
+
+In order to ensure data integrity of the non-initialized data when the chip
+reboots after a crash, it's important to flush the D-cache. The CMSIS API to
+perform that flush is:
+
+```c
+SCB_CleanDCache();
```
Comment (noahp) [inline docs/mcu/arm-mcuxpresso-guide.mdx:376]: applies to any Cortex-M7, might be something we could hook into generic docs or the SDK itself

Comment (chrisc11) [reply docs/mcu/arm-mcuxpresso-guide.mdx:376]: It's a CMSIS function so we could probably auto enable behind a __DCACHE_PRESENT check for default port

File: docs/mcu/arm-mcuxpresso-guide.mdx [hunk 2]
```diff
@@ -357,3 +357,29 @@ implementation.
...
+
+The Cortex-M7 chip on the i.MX RT10xx series chips has an instruction and data
+cache. The Memfault SDK relies on a minimum of 1 variable stored in
+non-initialized (`.noinit`) RAM when the system encounters a crash. If the
+RAM-backed coredump storage Memfault component is used, that data also should be
+located in a non-initialized section of RAM.
+
+In order to ensure data integrity of the non-initialized data when the chip
+reboots after a crash, it's important to flush the D-cache. The CMSIS API to
+perform that flush is:
```
Comment (chrisc11) [inline docs/mcu/arm-mcuxpresso-guide.mdx:373]: Maybe montiong from where it should be called? (i.e before or after reboot)

**Review by chrisc11 — APPROVED** [review]

File: docs/mcu/arm-mcuxpresso-guide.mdx (hunk 2 above)
Comment (noahp) [reply docs/mcu/arm-mcuxpresso-guide.mdx:373]: oh yeah good call :ok_hand:

**Pull request merged**